sponsorship_compassion/models/contracts.py

```diff
@@ -805,12 +824,6 @@ def _on_sponsorship_finished(self):
                     ('object_id', '=', sponsorship.id),
                 ]).unlink()
 
-        # Update sponsorships on partner

```

Comment:
Instead of having the same code twice in the `contract_cancelled` and `contract_terminated` methods, you can move your computation here in the `_on_sponsorship_finished` method.